### PR TITLE
storage docs

### PR DESCRIPTION
**Pull request by inian**
- [ ] guides
- [ ] blog post
- [ ] reference docs (in progress)

### CODE REVIEW

File: spec/supabase_js_v2.yml [hunk 1]
```diff
@@ -5334,6 +5334,44 @@ pages:
             .download('folder/avatar1.png')
           ```
 
+  storage.from.publicDownload():
```
Comment (dannykng) [inline spec/supabase_js_v2.yml:5337]: We'll want to add/move these to the `feat/docs2.0` branch in these files:
- https://github.com/supabase/supabase/blob/feat/docs2.0/spec/supabase_js_v2_temp_new_shape.yml
- https://github.com/supabase/supabase/blob/feat/docs2.0/spec/common-client-libs-sections.json

cc: @MildTomato @saltcod

File: spec/supabase_js_v2.yml [hunk 2]
```diff
@@ -5334,6 +5334,44 @@ pages:
...
+    title: 'from.publicDownload()'
+    $ref: '@supabase/storage-js.packages/StorageFileApi.default.publicDownload'
+    notes: |
+      - RLS policy permissions required1:
```
Comment (dannykng) [inline spec/supabase_js_v2.yml:5341]:
```suggestion
      - RLS policy permissions required:
```

File: spec/supabase_js_v2.yml [hunk 3]
```diff
@@ -5334,6 +5334,44 @@ pages:
...
+            .storage
+            .from('avatars')
+            .publicDownload('folder/avatar1.png')
+          ```
+
+  storage.from.authenticatedDownload():
+    title: 'from.authenticatedDownload()'
+    $ref: '@supabase/storage-js.packages/StorageFileApi.default.authenticatedDownload'
+    notes: |
+      - RLS policy permissions required1:
```
Comment (dannykng) [inline spec/supabase_js_v2.yml:5360]:
```suggestion
      - RLS policy permissions required:
```

File: spec/supabase_js_v2.yml [hunk 4]
```diff
@@ -5334,6 +5334,44 @@ pages:
...
           ```
 
+  storage.from.publicDownload():
+    title: 'from.publicDownload()'
+    $ref: '@supabase/storage-js.packages/StorageFileApi.default.publicDownload'
+    notes: |
+      - RLS policy permissions required1:
+        - `buckets` table permissions: none
+        - `objects` table permissions: `select`
+      - Refer to the [Storage guide](/docs/guides/storage#access-control) on how access control works
```
Comment (dannykng) [inline spec/supabase_js_v2.yml:5344]: Ah, I broke this section out into a separate doc so there are probably other links I need to update too.
```suggestion
      - Refer to the [Storage guide](/docs/guides/storage/access-control) on how access control works
```

File: spec/supabase_js_v2.yml [hunk 5]
```diff
@@ -5334,6 +5334,44 @@ pages:
...
+          ```
+
+  storage.from.authenticatedDownload():
+    title: 'from.authenticatedDownload()'
+    $ref: '@supabase/storage-js.packages/StorageFileApi.default.authenticatedDownload'
+    notes: |
+      - RLS policy permissions required1:
+        - `buckets` table permissions: none
+        - `objects` table permissions: `select`
+      - Refer to the [Storage guide](/docs/guides/storage#access-control) on how access control works
```
Comment (dannykng) [inline spec/supabase_js_v2.yml:5363]:
```suggestion
      - Refer to the [Storage guide](/docs/guides/storage/access-control) on how access control works
```

**Comment by inian** [comment]
will open a new PR based on the new docs site

**Pull request closed without merging**